### PR TITLE
refactor(levm): refactor opcodes

### LINKED ISSUE #1380: refactor(levm): opcode EXP gas consumption
The gas cost function implementation can be simplified very much if we pass the exponent as `U256` and use the `U256::bits()` in it.

### PR DESCRIPTION
**Pull request by damiramirez**
**Motivation**

In this PR I refactored some opcodes

**Description**

- `op_exp`: Move the conversion logic from `op_exp()` to `gas_cost::exp()`. Now `exp()` expects `U256` instead of `u64`.
- `op_mulmod`: Add early return when `multiplicand` or `multiplier` are zero.
- `op_addmod`: Add early return when `modulus` is zero.
- `op_div`: Remove redundant checking.
- `op_smod`: Add early return when `dividend` is zero.
- `op_sdiv`: Remove logic to check if the number is negative and use the `abs()` function
- `op_push`: I move the bytecode read and create the byte slice logic to helper functions. Then change the sign of `op_push` function, now it expect the number of bytes to move: `pub fn op_push(&mut self, current_call_frame: &mut CallFrame, n_bytes: usize) -> Result<OpcodeSuccess, VMError>`. TBH, I don't know if this is enough to simplify the func or maybe we can change something else. What do you think about this?

Closes #1380, #1381, #1382, #1383, #1384 and #1385

### CODE REVIEW

File: crates/vm/levm/src/opcode_handlers/arithmetic.rs [hunk 1]
```diff
@@ -215,12 +200,7 @@ impl VM {
         let base = current_call_frame.stack.pop()?;
         let exponent = current_call_frame.stack.pop()?;
 
-        let exponent_bits: u64 = exponent
-            .bits()
-            .try_into()
-            .map_err(|_| VMError::Internal(InternalError::ConversionError))?;
-
-        let gas_cost = gas_cost::exp(exponent_bits).map_err(VMError::OutOfGas)?;
+        let gas_cost = gas_cost::exp(exponent).map_err(VMError::OutOfGas)?;
```
Comment (ilitteri) [inline crates/vm/levm/src/opcode_handlers/arithmetic.rs:203]: If I'm not mistaken, we can remove this map.
```suggestion
        let gas_cost = gas_cost::exp(exponent)?;
```

File: crates/vm/levm/src/gas_cost.rs [hunk 2]
```diff
@@ -158,7 +158,12 @@ pub const CALLDATA_COST_NON_ZERO_BYTE: U256 = U256([16, 0, 0, 0]);
 // Blob gas costs
 pub const BLOB_GAS_PER_BLOB: U256 = U256([131072, 0, 0, 0]);
 
-pub fn exp(exponent_bits: u64) -> Result<U256, OutOfGasError> {
+pub fn exp(exponent: U256) -> Result<U256, OutOfGasError> {
```
Comment (ilitteri) [inline crates/vm/levm/src/gas_cost.rs:161]: I think we can do even better in this function. If we make `exponent_bits` an `U256` then we can avoid the `ConversionError` (and remove its declaration in `errors.rs`), then operate with `U256`s to finally get `exponent_byte_size` as `U256`, avoiding the `.into()` done afterwords (as it is already an `U256`).

**Review by ilitteri — COMMENTED** [review]
The PR looks good. Left two comments.

File: crates/vm/levm/src/opcode_handlers/arithmetic.rs (hunk 1 above)
Comment (damiramirez) [reply crates/vm/levm/src/opcode_handlers/arithmetic.rs:203]: Yep, you're right! Ty.

**Review by maximopalopoli — APPROVED** [review]
LGTM after attending Ivan's requests

File: crates/vm/levm/src/gas_cost.rs (hunk 2 above)
Comment (damiramirez) [reply crates/vm/levm/src/gas_cost.rs:161]: I changed the type to `U256` and then I used the `checked_div` in `exponent_byte_size`.

Comment (ilitteri) [reply crates/vm/levm/src/gas_cost.rs:161]: Great! We'll keep it this way for now, but take into account that divisions with constant non-zero divisors do not strictly need to be checked.

**Review by ilitteri — APPROVED** [review]

**Pull request merged**